### PR TITLE
Deprecate Component#isVisible

### PR DESCRIPTION
**Pull request by Parrryy**
[Rendered](https://github.com/Parrryy/rfcs/blob/44253a166ad12b7cd538eb818a5a1aa82fe5aec7/text/0000-deprecate-component-isvisible.md)

### CODE REVIEW

File: text/0000-deprecate-component-isvisible.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+An alternative option would be to change the `isVisible` property to use the
+hidden attribute of the element instead of adding the inline style.
+
+# Unresolved questions
+
+What would be the suggested way of hiding components from the DOM in Ember as
+there are a few options to choose from?
+
+When will support for the deprecated property end?
```
Comment (simonihmig) [inline text/0000-deprecate-component-isvisible.md]: It is public, so according to SemVer this should be 4.0

**Comment by Gaurav0** [comment]
As to when the support will end, I think since isVisible is public it will have to be 4.0

**Comment by Parrryy** [comment]
[New Draft](https://github.com/emberjs/rfcs/blob/5875aa0a60df9451255461b938e246fb8d88f4b3/text/0000-deprecate-component-isvisible.md) Updated the support ending section.

File: text/0000-deprecate-component-isvisible.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+In cases where the `isVisible` property is used we provide a deprecation warning
+with a link to the deprecation guide which states why it was deprecated and/or the
+options available to hide the component. As it is a public API it will be removed
+in the next major version release (4.0).
+
+There are several options available to hiding elements 
+such as `<div hidden={{boolean}}></div>`(hidden is valid for all elements
+and is semantically correct) or wrapping the component in a template
+conditional `{{#if}}`statement which do not interfere with
+the `StyleBindingReference`. Components `classNames` an `classNameBindings`
```
Comment (Serabe) [inline text/0000-deprecate-component-isvisible.md]: s/an/and/

**Comment by Parrryy** [comment]
[New Draft](https://github.com/emberjs/rfcs/blob/df29031f649417980c3f0cfd1da620dd828dcb3e/text/0000-deprecate-component-isvisible.md)

**Comment by MelSumner** [comment]
If anyone is curious about the accessibility concerns for display:none; - https://webaim.org/techniques/css/invisiblecontent/

**Comment by rwjblue** [comment]
This RFC came up at the Ember.js Core Team meeting today, and we have decided to move it forward into final comment period...

**Comment by rwjblue** [comment]
We reviewed this again at todays core team meeting, and are all still in favor. 

Thanks to everyone who participated!

**Pull request merged**